### DISCUSSION TITLE
Confusion on link checking/resolving

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3312] [Q&A] [answered]
The new link checks confuse me quite a bit...

The reason for this confusion is this:
I have a file inside `docs/posts/2020` called `12-24-truth-and-lies.md` which has a relative link `../12-20-in-defense-of-spigotmc`. This file also exists and is in the same directory.
However, MkDocs sees this relative link as a "unrecognized" link and gives the following line:
```
INFO    -  Doc file 'posts/2020/12-24-truth-and-lies.md' contains an unrecognized relative link '../12-20-in-defense-of-spigotmc', it was left as is.
```
The link itself works without any issues.
Now, changing it to have `.md` at the end makes things even weirder. Because now it gives a warning with the following:
```
WARNING -  Doc file 'posts/2020/12-24-truth-and-lies.md' contains a relative link '../12-20-in-defense-of-spigotmc.md', but the target 'posts/12-20-in-defense-of-spigotmc.md' is not found among documentation files.
```
What confuses me so much about this line is the target it gives, which is `posts/12-20-in-defense-of-spigotmc.md` which is a completely wrong path, especially given that the final link itself is pointing in the right direction which is `posts/2020/12-20-in-defense-of-spigotmc.md`
The link results in a 404 because of the .md extension and probs because MkDocs doesn't parse it due to the faulty path it has.

Is that an oversight? Because by common logic should `../12-20-in-defense-of-spigotmc.md` in `12-24-truth-and-lies.md` resolve like this
```
/posts/2020/12-24-truth-and-lies -> /posts/2020/ -> /posts/2020/12-20-in-defense-of-spigotmc
```

I'm so confused on what is happening here...

**Comment by Andre601** [comment]
Another weird behaviour I have is, that MkDocs reports a unrecognized relative link twice for one file, despite only one such link being in the entire file itself...

**Reply by Andre601** [reply]
Actually, looking closer at this it seems like it just reports one link twice for every file, which is quite weird.

**Reply by oprypin** [reply]
@Andre601 if you could share the site where this happens, that would be helpful. I'm not aware of it just reporting one link twice per file.

**Reply by Andre601** [reply]
Source is here: https://codeberg.org/Andre601/blog-source/

Note that it did show for me on local during serve and the stuff has been fixed (See selected answer).

Tho, I also use a CI where this has been shown in. Here's the link to the last build with these duplicate lines: https://ci.codeberg.org/repos/12455/pipeline/31/3

**Comment by kamilkrzyskow** [comment, marked as answer]
You're thinking in "absolute url directory" links, instead of file system paths. 
The file `docs/posts/2020/12-24-truth-and-lies.md` having a link `../12-20-in-defense-of-spigotmc` points to `docs/posts/12-20-in-defense-of-spigotmc`, because `..` go up 1 directory. It works in the website because the directory link would be `.com/posts/2020/12-24-truth-and-lies/` notice the `/` at the end, so the `..` correctly moves back to `2020` in the url, not in the file system. 

The warning about the lack of the `.md` extension is of course self-explanatory, because there is no file without the `.md`.  

So to fix the link in that file you should make it `12-20-in-defense-of-spigotmc.md` without the `../`

**Reply by Andre601** [reply]
Alright, thanks for the info.

Tho, one thing I found somewhat strange is, that one particular path was giving a Warning, while any other link of the same structure only gave the info line.
Like it was `../page` yet it gave a warn while a different file with another path but still `../file` gave a info line.

Could it be because the file name contained dots? It ended with `3.0.0.md` so the path had `3.0.0` at the end.
It's fixed now, but still weird...
